jib-core/src/main/java/com/google/cloud/tools/jib/blob/Blobs.java

```diff
@@ -79,11 +80,13 @@ public static String writeToString(Blob blob) throws IOException {
   }
 
   /**
-   * Gets a {@link Blob} that is {@code blob} compressed.
+   * Gets a {@link Blob} that is {@code blob} compressed. Note that the output stream is closed when
+   * the blob is written.
```

Comment:
Hmm... I don't quite like having an exception. It makes sense to not close an externally-owned stream. For example, some class may establish a network connection and write multiple blobs (in addition to other things) over the wire during the course. They may find it surprising if writing some opaque blob abruptly shuts down their output channel and renders the connection broken.

I think we can leave the output stream open by not using the try-resource block, simple as that. The owner of the output stream should eventually close it anyway.

Or, we can just move this function back to the test code and forget about any concern. The code is used only in tests, as you noticed.

Reply:
That last point was mostly my intent, hence the added `@VisibleForTesting`. Do you think we should just duplicate the method in the two test classes that use it like we had before?

Reply:
Yeah, I think that's fine. We can be sure the method is not used in prod.